cg/models/deliverables/metric_deliverables.py

```diff
@@ -119,6 +122,29 @@ def set_sample_ids(cls, _, values: dict) -> set:
         return set(sample_ids)
 
 
+class ConditionMetricsDeliverables(BaseModel):
+    """Specification for a metric deliverables file with conditions sets."""
+
+    metrics_: List[MetricsBase] = Field(..., alias="metrics")
+
+    @validator("metrics_", always=True)
+    def qc_pass(cls, metrics: List[MetricsBase]) -> str:
+        """Verify that metrics met QC conditions."""
+        failed_metrics: List = []
+        for metric in metrics:
+            if metric.condition is not None:
+                try:
+                    qc_function: Callable = getattr(operator, metric.condition.norm)
+                except AttributeError as error:
+                    raise CgError(
+                        f"{metric.condition.norm} is not an accepted operator for QC metric conditions."
+                    ) from error
+                if not qc_function(metric.value, metric.condition.threshold):
+                    failed_metrics.append(f"{metric.name}: {metric.value}")
+        if len(failed_metrics) != 0:
+            raise MetricsQCError(f"{';'.join(['QC failed'] + failed_metrics)}")
```

Comment:
```suggestion
            LOG.error(f"{';'.join(['QC failed'] + failed_metrics)}")
            raise MetricsQCError()
```

Reply:
I don't want to log the error here. I want to raise it with a given message

Reply:
You're right, you're expecting them upstream. 🥳 

Just in case you haven't consider it: What about the validation errors in case some of the required fields of `MetricsBase` are missed for example? Do you want to capture them also there or is it fine to have a `ValidationError`?

Reply:
Maybe you want to add them here:

```
    except MetricsQCError as error:
        analysis_api.trailblazer_api.set_analysis_status(
            case_id=case_id, status=AnalysisStatus.FAILED
        )
        analysis_api.trailblazer_api.add_comment(case_id=case_id, comment=error)
```

Reply:
This shouldn't happen. I'm thinking it's fine to have a validation error

Reply:
👍

Reply:
I did realize that I'm not verifying that all mandatory metrics are present. So I added an additional method

Reply:
I didn't managed in this PR. but you are right. I will add this info in the refactor issue. Resolving for now